jazz_events/index.js

```diff
@@ -40,382 +40,449 @@ module.exports.handler = (event, context, cb) => {
+			resolve(result);
+		})
+		.catch(function(error){
+			reject(error);
+		})
+	});
+}
 
-				if (err) {
-					logger.error(JSON.stringify(err));
-					if (err.code !== undefined && err.code === 500) {
-						cb(JSON.stringify(errorHandler.throwInternalServerError("An internal error occured. message: " + err.message)));
-					} else {
-						logger.error(JSON.stringify(err));
-						cb(JSON.stringify(errorHandler.throwInputValidationError("Bad request. message: " + err.message)));
-					}
+function validateEventHandler(config, event, dynamodb){
+	logger.info("Inside validateEventHandler:");
+	return new Promise((resolve, reject) => {
+		var event_handler_params = {
+			Key: {
+				"EVENT_HANDLER": {
+					S: event.body.event_handler
+				}
+			},
+			TableName: config.event_handler_table
+		};
+		getDynamodbItem(dynamodb, event_handler_params, event.body.event_handler)
+		.then(function(result){
+			resolve(result);
+		})
+		.catch(function(error){
+			reject(error);
+		})
+	});
+}
 
-				} else {
-					cb(null, responseObj(results.store_context, event.body));
+function validateEventStatus(config, event, dynamodb){
+	logger.info("Inside validateEventStatus:");
+	return new Promise((resolve, reject) => {
+		var event_status_params = {
+			Key: {
+				"EVENT_STATUS": {
+					S: event.body.event_status
 				}
+			},
+			TableName: config.event_status_table
+		};
+		getDynamodbItem(dynamodb, event_status_params, event.body.event_status)
+		.then(function(result){
+			resolve(result);
+		})
+		.catch(function(error){
+			reject(error);
+		})
+	});
+}
 
+function validateTimestamp(event){
+	logger.info("Inside validateTimestamp:");
+	return new Promise((resolve, reject) => {
+		try {
+			if (moment(event.body.event_timestamp, "YYYY-MM-DDTHH:mm:ss:SSS", true).isValid()) {
+				resolve(event.body.event_timestamp);
+			} else {
+				reject({
+					"code": 400,
+					"message": "Invalid EVENT TIMESTAMP: " + event.body.event_timestamp + ", The format should be YYYY-MM-DDTHH:mm:ss:SSS"
+				});
+			}
+		} catch (err) {
+			reject({
+				"code": 500,
+				"message": "Error parsing EVENT TIMESTAMP: " + event.body.event_timestamp + ", The format should be YYYY-MM-DDTHH:mm:ss:SSS"
 			});
 		}
+	});
+}
 
-	} catch (e) {
-		logger.error(e);
-		cb(JSON.stringify(errorHandler.throwInternalServerError(e)));
+function storeEventData(config, event, kinesis){
+	logger.info("Inside storeEventData:");
+	return new Promise((resolve, reject) => {
+		var event_id = Uuid();
+		var timestamp = moment().utc().format('YYYY-MM-DDTHH:mm:ss:SSS');
+		var event_params = {
+			Item: {
+				"EVENT_ID": {
+					S: event_id
+				},
+				"TIMESTAMP": {
+					S: timestamp
+				}
+			}
+		};
 
-	}
+		Object.keys(event.body).forEach(function (key) {
+			if (key === "service_context") {
+				event_params.Item.SERVICE_CONTEXT = {
+					S: JSON.stringify(event.body.service_context)
+				};
+			} else if (key === "event_handler") {
+				event_params.Item.EVENT_HANDLER = {
```

Comment:
You can reduce this if/else by using a hashmap. As an example:

let map = {'event_name': EVENT_NAME}
event_params.Item[map[key]] = { S: event.body[key]}
This might make your code more  concise rather than a big else if. What do you think?

Reply:
Updated as suggested.